lib/_http_outgoing.js

```diff
@@ -1046,6 +1044,14 @@ OutgoingMessage.prototype.end = function end(chunk, encoding, callback) {
     throw new ERR_HTTP_CONTENT_LENGTH_MISMATCH(this[kBytesWritten], this._contentLength);
   }
 
+  if (!this._headerSent && Buffer.byteLength(chunkBuffer)) {
+    this._send(NumberPrototypeToString(Buffer.byteLength(chunkBuffer), 16), 'latin1', null);
+    this._send(crlf_buf, null, null);
+    this._send(chunkBuffer, null, null);
+    this._send(crlf_buf, null, callback);
+    chunkBuffer = Buffer.alloc(0)
+  }
```

Comment:
This should be done in nextTick (i.e., connectionCorkNT) otherwise, there is no point to all of this...  the number of buffered chunks would always be 1.